### PR TITLE
Address issue 2652: not checking for host init in CUDA

### PR DESCRIPTION
**Pull request by crtrott**
This is for a specific way of how Legion wants to interact with Kokkos

### CODE REVIEW

**Comment by crtrott** [comment]
One possible way of addressing #2652

**Comment by crtrott** [comment]
Retest this please

**Comment by masterleinad** [comment]
Retest this please.

**Comment by crtrott** [comment]
Retest this please

**Comment by crtrott** [comment]
Any idea why this failed again @dalg24 @masterleinad

**Comment by crtrott** [comment]
Retest this please

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp [hunk 1]
```diff
@@ -313,6 +313,7 @@ void CudaInternal::initialize(int cuda_device_id, cudaStream_t stream) {
 
   enum { WordSize = sizeof(size_type) };
 
+#ifndef KOKKOS_IMPL_TURN_OFF_CUDA_HOST_INIT_CHECK
```
Comment (dalg24) [inline core/src/Cuda/Kokkos_Cuda_Instance.cpp]: So the intention is you configure with `-CMAKE_CXX_FLAGS="-D KOKKOS_IMPL_TURN_OFF_CUDA_HOST_INIT_CHECK"`?

Comment (DavidPoliakoff) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp]: Yeah, sorry, Christian and I discussed that on Slack

**Comment by streichler** [comment]
Confirming that this works for the current Legion/Kokkos interop code.  I'd prefer a longer term solution that doesn't involve knowing about (or anticipating) Legion+Kokkos usage when installing a shared Kokkos package on a system, but this unblocks things for now.

**Pull request merged**